<<<<<<< HEAD
From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
=======
From 3d43acc6068f00dbfc0c9a06355e2c8f7d302d0f Mon Sep 17 00:00:00 2001
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
From: Eneas U de Queiroz <cote2004-github@yahoo.com>
Date: Thu, 27 Sep 2018 08:30:24 -0300
Subject: Add openwrt targets

Targets are named: linux-$(CONFIG_ARCH)-openwrt

Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>

<<<<<<< HEAD
--- /dev/null
+++ b/Configurations/25-openwrt.conf
@@ -0,0 +1,59 @@
=======
diff --git a/Configurations/25-openwrt.conf b/Configurations/25-openwrt.conf
new file mode 100644
index 0000000000..86a86d31e4
--- /dev/null
+++ b/Configurations/25-openwrt.conf
@@ -0,0 +1,48 @@
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
+## Openwrt "CONFIG_ARCH" matching targets.
+
+# The targets need to end in '-openwrt' for the AFALG patch to work
+
+my %targets = (
+    "openwrt" => {
+	template	=> 1,
+	CFLAGS		=> add("\$(OPENWRT_OPTIMIZATION_FLAGS)"),
+    },
+    "linux-aarch64-openwrt" => {
+        inherit_from    => [ "linux-aarch64", "openwrt" ],
+    },
+    "linux-arc-openwrt" => {
<<<<<<< HEAD
+        inherit_from    => [ "linux-latomic", "openwrt" ],
=======
+        inherit_from    => [ "linux-generic32", "openwrt" ],
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
+    },
+    "linux-arm-openwrt" => {
+        inherit_from    => [ "linux-armv4", "openwrt" ],
+    },
+    "linux-armeb-openwrt" => {
+        inherit_from    => [ "linux-armv4", "openwrt" ],
+    },
+    "linux-i386-openwrt" => {
+        inherit_from    => [ "linux-x86", "openwrt" ],
+    },
<<<<<<< HEAD
+    "linux-loongarch64-openwrt" => {
+        inherit_from    => [ "linux64-loongarch64", "openwrt" ],
+    },
=======
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
+    "linux-mips-openwrt" => {
+        inherit_from    => [ "linux-mips32", "openwrt" ],
+    },
+    "linux-mips64-openwrt" => {
+        inherit_from    => [ "linux64-mips64", "openwrt" ],
+    },
+    "linux-mips64el-openwrt" => {
+        inherit_from    => [ "linux64-mips64", "openwrt" ],
+    },
+    "linux-mipsel-openwrt" => {
+        inherit_from    => [ "linux-mips32", "openwrt" ],
+    },
+    "linux-powerpc-openwrt" => {
+        inherit_from    => [ "linux-ppc", "openwrt" ],
+    },
<<<<<<< HEAD
+    "linux-powerpc64-openwrt" => {
+        inherit_from    => [ "linux-ppc64", "openwrt" ],
+        perlasm_scheme  => "linux64v2",
+    },
+    "linux-riscv64-openwrt" => {
+        inherit_from    => [ "linux-generic64", "openwrt" ],
+        perlasm_scheme   => "linux64",
+    },
=======
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
+    "linux-x86_64-openwrt" => {
+        inherit_from    => [ "linux-x86_64", "openwrt" ],
+    },
+
+### Basic default option
+    "linux-generic32-openwrt" => {
+        inherit_from    => [ "linux-generic32", "openwrt" ],
+    },
+);
